src/instrument.js

```diff
@@ -0,0 +1,44 @@
+import { decorate } from './private/utils';
+
+const CONSOLE_NATIVE = console;
+
+let count = 0;
+
+function handleDescriptor(target, key, descriptor, [prefix = null, options = {}]) {
+  let className = target.constructor.name;
+  let fn = descriptor.value;
+  const CONSOLE = options.console || CONSOLE_NATIVE;
+
+  if ( prefix === null ) {
+    prefix = `${className}.${key}`;
+  }
+
+  if (typeof fn !== 'function') {
+    throw new SyntaxError(`@instrument can only be used on functions, not: ${fn}`);
+  }
+
+  return {
+    ...descriptor,
+    value() {
+      let timeStarted;
+      let log = CONSOLE.log ? CONSOLE.log.bind(CONSOLE) : CONSOLE_NATIVE.log.bind(CONSOLE_NATIVE);
+      let time = CONSOLE.time ? CONSOLE.time.bind(CONSOLE) : function() {
+        timeStarted = new Date();
+      };
+      let timeEnd = CONSOLE.timeEnd ? CONSOLE.timeEnd.bind(CONSOLE) : function() {
```

Comment:
Makes sense. The suppressWarnings decorator tests might give you a reference https://github.com/jayphelps/core-decorators.js/blob/master/test/unit/suppress-warnings.spec.js